File: COT_TOT_PYTHON/agent4_results_cot/mlib_agent4_output_code/bug12.py
"""
Buggy NON-MODULAR snippet from matplotlib, bug ID 12.
"""
def buggy_function(rs, data, freq=None):
    y, xmin, xmax = cbook.delete_masked_points(y, xmin, xmax)
    xmin = np.resize(xmin, y.shape)
    xmax = np.resize(xmax, y.shape)
    verts = [((thisxmin, thisy), (thisxmax, thisy))
               for thisxmin, thisxmax, thisy in zip(xmin, xmax, y)]
    lines = mcoll.LineCollection(verts, colors=colors,
    x, ymin, ymax = cbook.delete_masked_points(x, ymin, ymax)
    ymin = np.resize(ymin, x.shape)
    ymax = np.resize(ymax, x.shape)
    verts = [((thisx, thisymin), (thisx, thisymax))
               for thisx, thisymin, thisymax in zip(x, ymin, ymax)]
    lines = mcoll.LineCollection(verts, colors=colors,
    return rs

File: COT_TOT_C/agent4_results_tot/mlib_agent4_output_code/bug12.c
```c
#include <stdio.h>
typedef void* Any;
Any buggy_function(Any rs, Any data, Any freq) {
    delete_masked_points(y, xmin, xmax);
    xmin = resize(xmin, y_shape);
    xmax = resize(xmax, y_shape);
    verts = build_verts(xmin, xmax, y);
    lines = line_collection(verts, colors);
    delete_masked_points(x, ymin, ymax);
    ymin = resize(ymin, x_shape);
    ymax = resize(ymax, x_shape);
    verts = build_verts_vertical(x, ymin, ymax);
    lines = line_collection(verts, colors);
    return rs;
}
```